azkaban-db/src/main/java/azkaban/db/DatabaseSetup.java

```diff
@@ -85,84 +61,38 @@ private void findMissingTables() {
         final String name = script.getName();
         final String[] nameSplit = name.split("\\.");
         final String tableName = nameSplit[1];
-        this.missingTables.add(tableName);
+        this.tables.add(tableName);
       }
     }
   }
 
-  private void createNewTables() throws SQLException, IOException {
+  private void createTables() throws SQLException, IOException {
     final Connection conn = this.dataSource.getConnection();
     conn.setAutoCommit(false);
     try {
-      // Make sure that properties table is created first.
-      if (this.missingTables.contains("properties")) {
-        runTableScripts(conn, "properties", this.version, this.dataSource.getDBType(),
-            false);
-      }
-      for (final String table : this.missingTables) {
-        if (!table.equals("properties")) {
-          runTableScripts(conn, table, this.version, this.dataSource.getDBType(), false);
-          // update version as we have create a new table
-          this.installedVersions.put(table, this.version);
-        }
+      for (final String table : this.tables) {
+        runTableScripts(conn, table);
```

Comment:
if one table creation throws an exception, should the rest fail or proceed?

Reply:
Yes, you should be right if we use this class for table create automation in production. This class is used for unit test only for now. We need to fix some reliability issues if we rely on this class to create tables when launching AZ in future. Updated java doc.

Reply:
Why is it better to proceed? Isn't it better to fail fast? This is used in tests.